src/features/projects/view/ProjectsContextProvider.tsx

```diff
@@ -11,9 +11,20 @@ const ProjectsContextProvider = ({
   initialProjects?: Project[],
   children?: React.ReactNode
 }) => {
+  const [refreshed, setRefreshed] = useState<boolean>(false)
   const [projects, setProjects] = useState<Project[]>(initialProjects || [])
+
+  const setProjectsAndRefreshed = (value: Project[]) => {
+    setProjects(value)
+    // Only set refreshed to true after projects are updated
+    if (JSON.stringify(projects) !== JSON.stringify(value)) setRefreshed(true)
```

Comment:
This check seems a little dangerous to me. Isn't there a risk of `JSON.stringify` generating different strings for the same value? For example, I think this could happen because key order doesn't matter in JSON, so I wouldn't expect `JSON.stringify` to generate strings with the same order of keys.

Reply:
That's a really good point, it could lead to false positives. I didn't think about that.
What do you think use a deep equal like isEqual from lodash ?

```suggestion
   if (projects.length !== value.length || !projects.every((project, index) => isEqual(project, value[index])))
```

Reply:
@mpabarca Sounds good to me. I'm not totally sure I understand why the check is needed, though. Are we explicitly calling this function too often and potentially supplying an old value?

Reply:
> @simonbs I found that the main issue causing the delay and showing error messages too early is that when we first pass the project data using `initialProjects`, it’s treated as cached data. This cached data is used to check if the project or branch exists.

That is a great observation! I didn't think of this at all in the PR I did earlier.

Reply:
> @simonbs new approach that compares specific fields first to ensure accuracy [...]

Sounds good to me. And should this prove to be insufficient, it's straightforward to add more properties.

> [...] and uses `JSON.stringify` as a fallback, reducing the risk of key order issues while maintaining reliable updates.

I'm unsure, but couldn't this be an issue still?

If I understand the solution correctly, I think that `JSON.stringify(projects) !== JSON.stringify(value)` can give false positives, i.e., the expression resolves to true even when there are no changes to the projects, which means that:

1. We think the data has been refreshed when it hasn't.
2. As a result, we might show error messages when we shouldn't.

I might not have the full picture here, so just let me know if this isn't actually a scenario 😊

Reply:
Hi @simonbs, after giving it more careful thought and reviewing all potential cases, relying on specific field comparisons (ID and versions) is enough to cover all scenarios, making JSON.stringify unnecessary and potentially counterproductive.
Let me know your feedback about it 😊

Reply:
@mpabarca Sounds good to me! Thanks for taking another look at this 🙏